apps/astarte_appengine_api/lib/astarte_appengine_api/kv_store.ex

```diff
@@ -54,13 +54,7 @@ defmodule Astarte.AppEngine.API.KvStore do
           Keyword.t()
         ) :: :ok | {:error, Exception.t()}
   def insert(kv_store_map, opts \\ []) do
-    %{
-      group: group,
-      key: key,
-      value: value
-    } = kv_store_map
-
-    value_type = Map.get(kv_store_map, :value_type, :binary)
+    value_type = Map.pop(kv_store_map, :value_type, :binary)
```

Comment:
Note that `Map.pop` returns `{value(), updated_map :: map()}` and not just the value directly.